src/pybind/rados/rados.pyx

```diff
@@ -4009,6 +4009,110 @@ returned %d, but should return zero on success." % (self.name, ret))
+        :param client: the client currently holding the lock
+        :param cookie: cookie of the lock
+
+        :raises: :class:`TypeError`
+        :raises: :class:`Error`
+        """
+        self.require_ioctx_open()
+
+        key_raw = cstr(key, 'key')
+        name_raw = cstr(name, 'name')
+        client_raw = cstr(client, 'client')
+        cookie_raw = cstr(cookie, 'cookie')
+
+        cdef:
+            char* _key = key_raw
+            char* _name = name_raw
+            char* _client = client_raw
+            char* _cookie = cookie_raw
+
+        with nogil:
+            ret = rados_break_lock(self.io, _key, _name, _client, _cookie)
+        if ret < 0:
+            raise make_ex(ret,
+                          "Ioctx.rados_break_lock(%s): failed to break lock %s "
+                          "on %s for client %s" % (self.name, name, key, client))
+
+    def list_lockers(self, key: str, name: str):
+
+        """
+        List clients that have locked the image and information
+        about the lock.
+
+        :param key: name of the object
+        :param name: name of the lock
+
+        :returns: dict - contains the following keys:
+
+                  * ``tag`` - the tag associated with the lock (every
+                    additional locker must use the same tag)
+                  * ``exclusive`` - boolean indicating whether the
+                     lock is exclusive or shared
+                  * ``lockers`` - a list of (client, cookie, address)
+                    tuples
+
+        :raises: :class:`TypeError`
+        :raises: :class:`Error`
+        """
+
+        self.require_ioctx_open()
+
+        key_raw = cstr(key, 'key')
+        name_raw = cstr(name, 'name')
+
+        cdef:
+            char* _key = key_raw
+            char* _name = name_raw
+            int exclusive = 0
+            size_t addrs_size = 512, tag_size = 512
+            size_t clients_size = 512, cookies_size = 512
+            char* c_tag = NULL
+            char* c_clients = NULL
+            char* c_cookies = NULL
+            char* c_addrs = NULL
+
+        try:
+            while True:
+                c_tag = <char *>realloc_chk(c_tag, tag_size)
+                c_cookies = <char *>realloc_chk(c_cookies, cookies_size)
+                c_clients = <char *>realloc_chk(c_clients, clients_size)
+                c_addrs = <char *>realloc_chk(c_addrs, addrs_size)
+                with nogil:
+                    ret = rados_list_lockers(self.io, _key, _name, &exclusive,
+                                             c_tag, &tag_size, c_clients, &clients_size,
+                                             c_cookies, &cookies_size, c_addrs, &addrs_size)
+                if ret >= 0:
+                    break
+                elif ret != -errno.ERANGE:
+                    raise make_ex(ret,
+                                  "Ioctx.rados_list_lockers(%s): failed to list lockers" % (self.name))
+            if ret == 0:
+                return []
```

Comment:
It's a bit weird, but I'm pretty sure that the RADOS lock continues to exist and maintains its exclusive disposition and tag even if there are no lockers.  In case `ret == 0`, I'd still return the same dictionary -- just with `lockers` list being empty.